bug-batch-oc: Bug batching PR for 4.19 (#94732)
openshift/openshift-docs: release_notes/ocp-4-19-release-notes.adoc

[id="ocp-release-note-observability-bug-fixes_{context}"]
==== Observability

[discrete]
[id="ocp-release-note-oc-cli-bug-fixes_{context}"]
==== OpenShift CLI (oc)

* Previously, when adding a node in a disconnected environment using the `oc adm node-image` command, private registry images were inaccessible to the command, causing node addition failure due to image pull issues. This error occurred only if the cluster was initially installed with an installer binary downloaded from mirror.openshift.com. With this release, a fix has been implemented that enables successful image pull and node creation in disconnected environments. (link:https://issues.redhat.com/browse/OCPBUGS-53106[OCPBUGS-53106])

* Previously, when mirroring Operators using oc-mirror plugin v2, some community Operators with long lists of `skips` and `replaces` entries in their channel graphs caused the mirroring process to run out of memory and fail.
+

> With this update, oc-mirror plugin v2 improves the filtering logic by avoiding repeated evaluation of entries referenced in multiple `skips` and `replaces` stanzas, resulting in better memory handling during Operator mirroring. (link:https://issues.redhat.com/browse/OCPBUGS-52471[OCPBUGS-52471])

Reviewer: should this bug be in the oc mirror section instead?

* Previously, a bug on `oc adm inspect --all-namespaces` command construction meant that must-gather was not correctly gathering information about leases, `csistoragecapacities`, and the assisted-installer namespace. With this release, the issue is fixed and must-gather will gather the information correctly. (link:https://issues.redhat.com/browse/OCPBUGS-44857[OCPBUGS-44857]) 

* Previously, the `oc adm node-image create --pxe generated` command did not create only the Preboot Execution Environment (PXE) artifacts. Instead, the command created the PXE artifacts with other artifacts from a `node-joiner` pod and stored them all in the wrong subdirectory. Additionally, the PXE artifacts were incorrectly prefixed with `agent` instead of `node`. With this release, generated PXE artifacts are stored in the correct directory and receive the correct prefix. (link:https://issues.redhat.com/browse/OCPBUGS-45311[OCPBUGS-45311])

[discrete]
[id="ocp-release-note-oc-mirror-bug-fixes_{context}"]
==== oc-mirror

> * Previously, oc-mirror plugin v2 did not display any progress output during the local cache population phase. For mirror configurations involving a large number of images, this could make the process appear unresponsive or stuck.

Reviewer: ```suggestion
* Previously, when mirroring Operators using oc-mirror plugin v2, some community Operators with long lists of `skips` and `replaces` entries in their channel graphs caused the mirroring process to run out of memory and fail. With this update, oc-mirror plugin v2 improves the filtering logic by avoiding repeated evaluation of entries referenced in multiple `skips` and `replaces` stanzas, resulting in better memory handling during Operator mirroring. (link:https://issues.redhat.com/browse/OCPBUGS-52471[OCPBUGS-52471])

* Previously, oc-mirror plugin v2 did not display any progress output during the local cache population phase. For mirror configurations involving a large number of images, this could make the process appear unresponsive or stuck. With this update, a progress bar has been added to provide visibility into the cache population status for user awareness and experience. (link:https://issues.redhat.com/browse/OCPBUGS-56563[OCPBUGS-56563])
```

> With this update, a progress bar has been added to provide visibility into the cache population status, for user awareness and experience. (link:https://issues.redhat.com/browse/OCPBUGS-56563[OCPBUGS-56563])

Reviewer: ```suggestion
```

> * Previously, when re-running oc-mirror plugin v2 with the same working directory, existing `tar` archive files from previous runs were not removed. This resulted in a mix of outdated and new archives, which could cause mirroring failures when pushing to the target registry.

Reviewer: ```suggestion
* Previously, when re-running oc-mirror plugin v2 with the same working directory, existing `tar` archive files from previous runs were not removed. This resulted in a mix of outdated and new archives, which could cause mirroring failures when pushing to the target registry. With this update, oc-mirror plugin v2 automatically deletes old `tar` archive files at the beginning of each run, ensuring that the working directory only contains archives from the current execution. (link:https://issues.redhat.com/browse/OCPBUGS-56433[OCPBUGS-56433])
```

+

> With this update, oc-mirror plugin v2 automatically deletes old `tar` archive files at the beginning of each run, ensuring that the working directory only contains archives from the current execution. (link:https://issues.redhat.com/browse/OCPBUGS-56433[OCPBUGS-56433])

Reviewer: ```suggestion
```

> * Previously, oc-mirror v2 would terminate with an error if the source registry responded with an HTTP 502, 503, or 504 status code during image copying.

Reviewer: ```suggestion
* Previously, oc-mirror plugin v2 would terminate with an error if the source registry responded with any of the following HTTP status codes during image copying: 502, 503, 504. With this update, oc-mirror plugin v2 automatically retries the copy operation when encountering these temporary server errors. (link:https://issues.redhat.com/browse/OCPBUGS-56185[OCPBUGS-56185])
```

> +

Reviewer: ```suggestion
```

> With this update, oc-mirror v2 now automatically retries the copy operation when encountering these temporary server errors. (link:https://issues.redhat.com/browse/OCPBUGS-56185[OCPBUGS-56185])

Reviewer: ```suggestion
```

* Previously, when mirroring a Helm chart that included container images with both a tag and digest in the reference, oc-mirror plugin v2 failed with the following error:
+
[source,text]
----
Docker references with both a tag and digest are currently not supported.
----
+

> With this update, oc-mirror plugin v2 supports Helm charts that reference images using both tag and digest. The tool mirrors the image using the digest as the source and applies the tag at the destination.

Reviewer: ```suggestion
With this update, oc-mirror plugin v2 supports Helm charts that reference images using both tag and digest. The tool mirrors the image using the digest as the source and applies the tag at the destination. (link:https://issues.redhat.com/browse/OCPBUGS-54891[OCPBUGS-54891])
```

> (link:https://issues.redhat.com/browse/OCPBUGS-54891[OCPBUGS-54891]) 

Reviewer: ```suggestion
```

> * Previously, during image cleanup, oc-mirror plugin v2 would stop the deletion process if any error occurred while removing an image.

Reviewer: ```suggestion
* Previously, during image cleanup, oc-mirror plugin v2 would stop the deletion process if any error occurred while removing an image. With this update, oc-mirror plugin v2 continues attempting to delete remaining images even after encountering errors. After the process completes, it displays a list of any failed deletions. (link:https://issues.redhat.com/browse/OCPBUGS-54653[OCPBUGS-54653])
```

> +

Reviewer: ```suggestion
```

> With this update, oc-mirror plugin v2 continues attempting to delete remaining images even after encountering errors. After the process completes, it displays a list of any failed deletions.

Reviewer: ```suggestion
```

> (link:https://issues.redhat.com/browse/OCPBUGS-54653[OCPBUGS-54653])

Reviewer: ```suggestion
```

> * Previously, it was possible to mirror an empty catalog during the mirror-to-disk (m2d) phase if an invalid Operator was specified in the `ImageSetConfiguration` file. This led to a failure during the subsequent disk-to-mirror (d2m) phase.

Reviewer: ```suggestion
* Previously, it was possible to mirror an empty catalog during the mirror-to-disk (m2d) phase if an invalid Operator was specified in the `ImageSetConfiguration` file. This led to a failure during the subsequent disk-to-mirror (d2m) phase. With this update, oc-mirror plugin v2 prevents mirroring empty catalogs by validating operator references in the configuration, ensuring a more reliable mirroring process (link:https://issues.redhat.com/browse/OCPBUGS-52588[OCPBUGS-52588])
```

> +

Reviewer: ```suggestion
```

> With this update, oc-mirror plugin v2 prevents mirroring empty catalogs by validating operator references in the configuration, ensuring a more reliable mirroring process.

Reviewer: ```suggestion
```

> (link:https://issues.redhat.com/browse/OCPBUGS-52588[OCPBUGS-52588])

Reviewer: ```suggestion
```

> * Previously, when using oc-mirror plugin v2 with the `--dry-run` flag, the `cluster-resources` folder inside the working directory was cleared. As a result, previously generated files such as `idms-oc-mirror.yaml` and `itms-oc-mirror.yaml` were deleted.

Reviewer: ```suggestion
* Previously, when using oc-mirror plugin v2 with the `--dry-run` flag, the `cluster-resources` folder inside the working directory was cleared. As a result, previously generated files such as `idms-oc-mirror.yaml` and `itms-oc-mirror.yaml` were deleted. With this update, the `cluster-resources` folder is no longer cleared during dry-run operations, preserving any previously generated configuration files. (link:https://issues.redhat.com/browse/OCPBUGS-50963[OCPBUGS-50963])
```

> +

Reviewer: ```suggestion
```

> With this update, the `cluster-resources` folder is no longer cleared during dry-run operations, preserving any previously generated configuration files. (link:https://issues.redhat.com/browse/OCPBUGS-50963[OCPBUGS-50963])

Reviewer: ```suggestion
```

> * Previously, the oc-mirror plugin v2 returned an exit status of `0` (success) even when mirroring errors occurred. As a result, oc-mirror plugin v2 run failures in automated workflows could go undetected.

Reviewer: ```suggestion
* Previously, the oc-mirror plugin v2 returned an exit status of `0` (success) even when mirroring errors occurred. As a result, oc-mirror plugin v2 run failures in automated workflows could go undetected. With this update, oc-mirror plugin v2 has been updated to return a non-zero exit status when mirror failures occur. Despite this fix, users should not rely solely on the exit status in automated workflows. Users are advised to manually check the `mirroring_errors_XXX_XXX.txt` file generated by oc-mirror plugin v2 to identify any potential issues. (link:https://issues.redhat.com/browse/OCPBUGS-49880[OCPBUGS-49880])
```

+

> With this update, oc-mirror plugin v2 has been updated to return a non -`0` exit status when mirror failures occur.

Reviewer: ```suggestion
```

> +

Reviewer: ```suggestion
```

> Despite this fix, users should not rely solely on the exit status in automated workflows. Users are advised to manually check the `mirroring_errors_XXX_XXX.txt` file generated by oc-mirror plugin v2 to identify any potential issues. (link:https://issues.redhat.com/browse/OCPBUGS-49880[OCPBUGS-49880])

Reviewer: ```suggestion
```

> * Previously, when mirroring using an internal oc-mirror reserved keyword—such as `release-images`—in the destination or `--from` path flags, the operation could fail or behave unexpectedly.

Reviewer: ```suggestion
* Previously, when mirroring using an internal oc-mirror reserved keyword, such as `release-images`, in the destination or `--from` path flags, the operation could fail or behave unexpectedly. With this update, oc-mirror plugin v2 correctly handles reserved keywords used in destination or source paths. (link:https://issues.redhat.com/browse/OCPBUGS-42862[OCPBUGS-42862])
```

+

> With this update, oc-mirror plugin v2 correctly handles reserved keywords used in destination or source paths.

Reviewer: ```suggestion
```

> (link:https://issues.redhat.com/browse/OCPBUGS-42862[OCPBUGS-42862])

Reviewer: ```suggestion
```
